rfcs/0000-dns-cluster-aware-delegation.md

```diff
@@ -0,0 +1,170 @@
+DNSRecords on these clusters must have access to valid provider credentials referenced in the spec.
+The authoritative DNSRecord is created in the same namespace as the original DNSRecord (created by the DNSPolicy or otherwise) and at a minimum will contain the primary DNSRecords endpoints.
+The remaining specification of the authoritative DNSRecord is synced from the original primary DNSRecord (rootHost, providerRef etc..) 
+This mode should be selected for multi cluster scenarios where a designated "control plane" cluster(s) are being configured with elevated privileges to access external DNS providers.
+
+### Remote
+
+DNSRecords will be reconciled by this instance, but only to maintain a finalizer on the resource. 
+The expectation is that a different "primary" cluster will reconcile the resource, adding its endpoints to the shared authoritative record, and update its status.
+
+Note: While this concept is intended to work with the multi cluster concept described below, it is also possible to use this on a single cluster to combine several records together. 
+This might be desirable if you have many records and want to reduce the number of API calls to a provider and also the number of owner TXT records that need to be managed.
+
+## Cluster Aware Controller
+
+The DNS reconciler can be made cluster aware allowing it to process DNSRecord resources on other clusters as well as its own.
+
+Making the controller aware of other clusters is done via specially labeled secret resources containing kubeconfig data that exist in the `kuadrant-system` namespace.
+
+The kubeconfig data should be generated from an appropriately scoped service account that exists on the target cluster.
+
+When a cluster secret is created, the DNS controller on that cluster will automatically detect it and start watching for DNSRecord resources on that external cluster as well as its own.
+
+**Example** Secret containing kubeconfig data
+```yaml
+apiVersion: v1
+kind: Secret
+data:
+  kubeconfig: <kubeconfig data>
+metadata:
+  labels:
+    kuadrant.io/multicluster-kubeconfig: "true"
+  name: kind-kuadrant-local-2
+  namespace: kuadrant-system
+type: Opaque
+```
+
+## Namespace requirements for multi cluster
+
+ToDo Expectations of layout of namespaces and names of DNSRecords to match across clusters
+
+## Example Configurations
+
+**Default configuration.** Multi cluster using default node (Current behaviour)
+
+![img.png](./0000-dns-cluster-aware-delegation-assets/multi-cluster-default-mode.png)
+
+**Primary/Remote Configuration.** Multi cluster using primary/remote modes and cluster aware controller
+
+![img.png](./0000-dns-cluster-aware-delegation-assets/multi-custer-primary-remote-cluster-aware.png)
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## CRD Provider
```

Comment:
I'm thinking this might be better called "dnsrecord" provider since that's what it actually targets, CRD might be suggesting it could be anything.